solr/distribution/build.gradle

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.lucene.gradle.Checksum
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+plugins {
+  id 'signing'
+}
+
+// This project puts together the Solr "distribution", assembling bits and pieces
+// from across the project structure into release artifacts.
+
+ext {
+  releaseDir = file("${buildDir}/release")
+  withSignedArtifacts = { ->
+    def propValue = propertyOrDefault("sign", null)
+    // Allow -Psign to work as a shorthand for -Psign=true
+    return propValue != null && (propValue.isBlank() || Boolean.parseBoolean(propValue))
+  }.call()
+  useGpgForSigning = { ->
+    def propValue = propertyOrDefault("useGpg", null)
+    // Allow -PuseGpg to work as a shorthand for -PuseGpg=true
+    return propValue != null && (propValue.isBlank() || Boolean.parseBoolean(propValue))
+  }.call()
+}
+
+if (project.ext.useGpgForSigning) {
+  signing {
+    useGpgCmd()
+  }
```

Comment:
Hmm, is this missing from the lucene build as well? I was pretty sure gpg signing worked for lucene?

Reply:
Aha, lucene has a `gradle/maven/singing.gradle` that basically just does this. Can copy that over instead

Reply:
Actually there are many more files in `gradle/maven` for lucene. Should we move a lot of them over?

Reply:
I guess we may need some more stuff from Lucene to do the actual maven publishing. @dweiss would know.

But that could be done in a new PR, right? Let's get this in so we get progress on building a RC, playing with docker builds etc.

Reply:
Here's the artifact-signing bit in Lucene:
https://github.com/apache/lucene/blob/main/lucene/distribution/artifact-signing.gradle